fix: di and config mgmt

jleach opened the pull request:
This PR makes the following changes:

* **Fixes #34**: I diffed the work done by [Credebl](https://github.com/credebl/mediator-agent) and this repo. Looks like the problem was caused some imports using the word `type`. I've removed all the ones causing problems and left the others.
* **Moved constants to config:** I moved some constants to a config file so they can be easily exposed in Helm charts. This allows Kubernetes to use environment variables for configuration.
* **Added `dotenv` support:** I added support for `dotenv` so that developers can use environment variables in development environments. In production, the library will load from `process.env.*`.
* **biome rules:** The biome rule [useImportType](https://biomejs.dev/linter/rules/use-import-type/) was causing issues so I disabled it.

jleach commented:
@TimoGlastra This runs fine. I still need to fix the style errors. Wanted to put it up for early review since I kinda went rogue in fixing #34. Thoughts?

TimoGlastra commented on src/config.ts:
```diff
@@ -0,0 +1,63 @@
+// nconf.defaults({});
+
+if (env === 'development') {
+  console.log('Configuration:', nconf.get());
```
Console.log

TimoGlastra reviewed (commented):
Generally ok, but could use some cleanup. 

Haven't used nconf before but i like the config file based approach.

sairanjit commented:
@jleach You can remove the `clientCode` as it was built for some custom logic and can be removed now.

jleach commented on src/agent.ts:
```diff
@@ -48,14 +50,14 @@ export async function createAgent() {
   const socketServer = new Server({ noServer: true })
 
-  const logger = new Logger(LOG_LEVEL)
+  const logger = new Logger(config.get('agent:logLevel'))
```
Something like:

```
import dotenv from "dotenv";

dotenv.config();

interface Config {
  PORT: number;
  NODE_ENV: string;
}

const config: Config = {
  PORT: parseInt(process.env.PORT || "3000", 10),
  NODE_ENV: process.env.NODE_ENV || "development",
};
```

jleach commented:
> This pr is undoing quite a few things i added. Can you check to only integrate needed changes from credebl repo?

For sure. That was just a draft for early feedback. I wen't through and diffed the two. Looks like #34 was caused by some `import { type X } from 'foo'` where the `type` was causing the problem. I went through and removed all the ones that were causing the errors.

This PR is mostly cosmetic now except for the `config` changes. I'm Ok with a more simple approact:

```
import dotenv from "dotenv";

if (env === 'development') {
  dotenv.config();
}

type Config = {
  PORT: number;
  NODE_ENV: string;
}

const config: Config = {
  PORT: parseInt(process.env.PORT || "3000", 10),
  NODE_ENV: process.env.NODE_ENV || "development",
};
```

Thoughts?

TimoGlastra commented:
This is the env feature in node 20.6+ https://dev.to/usulpro/nodejs-a-guide-to-native-env-support-and-local-development-300g

TimoGlastra commented:
Re the biome rule, somehow it doesn't recognize the service is being used for dependency injection. It shouldn't be type import there. Disabling the rule makes sense for now

sairanjit commented on package.json:
```diff
@@ -18,7 +18,7 @@
     "build": "tsc -p tsconfig.build.json",
     "prepublishOnly": "pnpm build",
-    "dev": "ts-node --env-file=.env.development --env-file .env.local dev",
+    "dev": "ts-node src/index.ts",
```
@jleach Do you want to add the config related changes to `dev.ts` as well ? As it helps if we want to run locally using `ngrok`

jleach replied:
I don't think you need `dev` any more. Just change the `AGENT_ENDPOINTS='https://mediator.example.ca,wss://mediator.example.ca'` in your .env file and `pnpm dev` and you're good to go. To me there shouldn't me any more of a change between dev and test. Thoughts?

TimoGlastra replied:
The dev script connected to ngrok and changed the urls. Now you have to run it manually in another terminal and set the env variables.

It's ok, but does require more setup work

jleach replied:
@TimoGlastra I don't mind. I can create create a `pnpm dev:standalone` or something for ppl that don't use ngrok. Will create a new ticket for tracking since I merged this one already.
